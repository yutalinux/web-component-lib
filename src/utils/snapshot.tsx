import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
