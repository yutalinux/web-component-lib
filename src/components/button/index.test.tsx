import { expect, test } from "bun:test";
import { snapshot } from "@/utils/tests/snapshot";
import Button from ".";

test("Button Snapshot", () => {
  expect(snapshot(Button, { label: "Button" })).toMatchSnapshot();
  expect(
    snapshot(Button, { color: "secondary", label: "Button" }),
  ).toMatchSnapshot();
  expect(
    snapshot(Button, { size: "small", label: "Button" }),
  ).toMatchSnapshot();
  expect(
    snapshot(Button, { size: "large", label: "Button" }),
  ).toMatchSnapshot();
});
