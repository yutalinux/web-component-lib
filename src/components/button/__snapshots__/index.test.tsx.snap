// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `"<button type="button" class="rounded-md outline-none text-white bg-black hover:bg-neutral-900 dark:bg-white dark:text-black dark:hover:bg-neutral-100 text-base px-2 py-1" data-rac="" id="react-aria-:R0:">Button</button>"`;

exports[`Button Snapshot 2`] = `"<button type="button" class="rounded-md outline-none border box-border text-black bg-white hover:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:hover:bg-neutral-800 text-base px-2 py-1" data-rac="" id="react-aria-:R0:">Button</button>"`;

exports[`Button Snapshot 3`] = `"<button type="button" class="rounded-md outline-none text-white bg-black hover:bg-neutral-900 dark:bg-white dark:text-black dark:hover:bg-neutral-100 text-sm px-2 py-1" data-rac="" id="react-aria-:R0:">Button</button>"`;

exports[`Button Snapshot 4`] = `"<button type="button" class="rounded-md outline-none text-white bg-black hover:bg-neutral-900 dark:bg-white dark:text-black dark:hover:bg-neutral-100 text-lg px-2 py-1" data-rac="" id="react-aria-:R0:">Button</button>"`;
